tutorials/QuantumClassification/QuantumClassificationWithFeatureEngineering.ipynb

```diff
@@ -0,0 +1,1048 @@
+    "}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### <span style=\"color:blue\">Exercise 4.</span> Figure out feature engineering parameters for classes separable by a hyperbola\n",
+    "\n",
+    "In the code cell below, replace `...` with the parameters $[a_0]$ or $[a_0, a_1]$ used with left padding preprocessing to produce the data for training and classification."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "invalid_code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# TODO: figure out parameters for left padding feature engineering\n",
+    "feature_engineering_params_hyperbola = [...]\n",
+    "\n",
+    "# Train the model\n",
+    "(parameters, bias) = TrainModelHyperbolaData.simulate(\n",
+    "    trainingVectors = training_data_hyperbola['Features'],\n",
+    "    trainingLabels = training_data_hyperbola['Labels'],\n",
+    "    featureEngineeringParameters = feature_engineering_params_hyperbola,\n",
+    "    initialParameters = [[1.0], [2.0], [3.0]]\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true,
+    "tags": [
+     "invalid_code"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Classify validation data set using training results\n",
+    "classified_labels = ClassifyHyperbolaData.simulate(\n",
+    "    samples = validation_data_hyperbola['Features'],\n",
+    "    featureEngineeringParameters = feature_engineering_params_hyperbola,\n",
+    "    parameters = parameters, bias = bias\n",
+    ")\n",
+    "\n",
+    "# Calculate miss rate\n",
+    "mask = np.not_equal(validation_data_hyperbola['Labels'], classified_labels)\n",
+    "miss_count = np.array(classified_labels)[np.where(mask)].size\n",
+    "miss_rate = miss_count / len(classified_labels)\n",
+    "print(f\"Miss rate: {miss_rate:0.2%}\")\n",
+    "\n",
+    "# Plot validation results\n",
+    "plot_data(validation_data_hyperbola['Features'], validation_data_hyperbola['Labels'], classified_labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What's Next?\n",
+    "\n",
+    "This tutorial covered classifying artificial data, taking advantage of its simple structure. Classifying real data will require more complex models - same as in traditional machine learning.\n",
+    "\n",
+    "* Check out [introduction to quantum machine learning](https://docs.microsoft.com/azure/quantum/user-guide/libraries/machine-learning/) at Microsoft Quantum Development Kit documentation, which features a more interesting example - classifying half-moons dataset.\n",
+    "* [Quantum machine learning samples](https://github.com/microsoft/Quantum/tree/main/samples/machine-learning) offer examples of classifying several more datasets."
+   ]
+  }
+ ],
+ "metadata": {
```

Comment:
Is this necessary?

Reply:
I'm not sure this comment ended up on the line it was intended for, metadata is a necessary part of a notebook, yes